pkg/catalogv2/content/content_test.go

```diff
@@ -0,0 +1,204 @@
+			">2.5.0-rc1 <=2.6.0",
+			"v2.6.0",
+			true,
+		},
+		{
+			"rancher version in range comparison with `>= <=`style comparison",
+			">=2.5.0-alpha2 <=2.6.0",
+			"v2.5.0",
+			true,
+		},
+		{
+			"rancher version in range comparison with `~` style comparison",
+			"~2.5.x", //equivalent to ">= 2.5.0 < 2.6.0"
+			"v2.5.7",
+			true,
+		},
+		{
+			"rancher version in range comparison with `<` style comparison",
+			"<2.6.0",
+			"v2.5.7",
+			true,
+		},
+		{
+			"rancher version in range comparison with `<=` style comparison",
+			"<= 2.5.8-rc7",
+			"v2.5.8-rc2+123",
+			true,
+		},
+		{
+			"rancher version in range comparison with `>=` style comparison",
+			">= 2.4.3",
+			"v3.0.0+123",
+			true,
+		},
+		{
+			"rancher version in range comparison with `>` style comparison",
+			">2.4.3",
+			"v2.4.4",
+			true,
+		},
+		{
+			"rancher version in range comparison with `-` style comparison",
+			"2.5 - 2.6.3", //equivalent to ">= 2.5 <= 2.6.3"
+			"v2.5.9",
+			true,
+		},
+		{
+			"rancher version in range comparison with `^` style comparison",
+			"^2.7.5", //equivalent to ">= 2.7.5, < 2.8.0"
+			"v2.7.8",
+			true,
+		},
+		{
+			"rancher version out of range comparison with `>= <`style comparison",
+			">= 2.5.0-alpha <2.6.0",
+			"v2.4.9",
+			false,
+		},
+		{
+			"rancher version out of range comparison with `> <`style comparison",
+			">2.5.0 <2.6.0",
+			"v2.5.3-alpha3", //pre-release version can only be used if on of the semver constrains has the same pre-release version
+			false,
+		},
+		{
+			"rancher version out of range comparison with `> <=`style comparison",
+			"> 2.5.0-rc2 <=2.6.0",
+			"v2.5.0-beta2", //pre-release version can only be used if on of the semver constrains has the same pre-release version
+			false,
+		},
+		{
+			"rancher version out of range comparison with `>= <=`style comparison",
+			">=2.5.0-rc1 <=2.6.0",
+			"v2.4.2",
+			false,
+		},
+		{
+			"rancher version out of range comparison with `~` style comparison",
+			"~2.5.x", //equivalent to >= 2.5.0, < 2.6.0
+			"v2.6.0",
+			false,
+		},
+		{
+			"rancher version out of range comparison with `<` style comparison",
+			"<2.6.0-alpha",
+			"v2.7.3",
+			false,
+		},
+		{
+			"rancher version out of range comparison with `<=` style comparison",
+			"<=2.6.0",
+			"v2.6.1",
+			false,
+		},
+		{
+			"rancher version out of range comparison with `>=` style comparison",
+			">= 2.4.3",
+			"v2.4.2-alpha1", //alpha can only be used if on of the semver constrains is also an alpha
+			false,
+		},
+		{
+			"rancher version out of range comparison with `>` style comparison",
+			">2.4.3",
+			"v2.4.3",
+			false,
+		},
+		{
+			"rancher version out range comparison with `-` style comparison",
+			"2.5 - 2.6.3", //equivalent to ">= 2.5 <= 2.6.3"
+			"v2.4.9",
+			false,
+		},
+		{
+			"rancher version out range comparison with `^` style comparison",
+			"^2.7.5", //equivalent to ">= 2.7.5 < 3.0.0"
+			"v3.0.0",
+			false,
+		},
```

Comment:
Out of curiosity I wonder about 2 cases: 

* One that is 000, the invalid semver, to see how it reacts. like `"2.5.000 - 2.6.005"`
* A dash metadata: '>=v1.14.0-r0'

Neither of those would likely end up as rancher version but wonder if they break this.

Reply:
I have added a few random scenarios for `000` and for random meta data `r8` - its all evaluated with ASCII sort order - https://github.com/Masterminds/semver/blob/49c09bfed6adcffa16482ddc5e5588cffff9883a/constraints_test.go#L262 << here are additional tests from the library. These are the tests for the specific methods we are calling to filter out our charts.